### ISSUE TITLE
Phase 5: Retrain surrogate with ASHRAE 140 validated physics

### ISSUE THREAD

**Issue by anchapin** [anchapin/fluxion#106]
## Summary

According to the roadmap, Phase 5 involves replacing synthetic training data with real data calibrated from ASHRAE 140 validation. The current surrogate was trained on simplified/synthetic data.

## Current State

- ✅ Surrogate training pipeline exists (`tools/train_surrogate.py`)
- ✅ ONNX model export works
- ✅ Basic validation framework exists
- ❌ Model trained on simplified/synthetic data, not ASHRAE-calibrated physics

### 1. Data Collection
- Generate training data from validated ASHRAE 140 cases
- Use the analytical physics model (not simplified version)
- Cover parameter space: u_value (0.5-3.0), setpoint (19-24°C)

### 2. Retraining Pipeline
```bash
# Generate training data from validated cases
python tools/data_gen/main.py --cases 600,900 --num-samples 50000

# Retrain model
python tools/train_surrogate.py --data assets/ashrae_calibrated_data.npz --epochs 100

# Export calibrated model
python -c "import torch; model = load('model.pt'); torch.onnx.export(model, 'assets/thermal_surrogate_v2.onnx')"
```

### 3. Validation Metrics
Target accuracy after retraining:
- MAE < 5% vs ASHRAE 140 reference
- RMSE < 8%
- Per-zone accuracy > 95%

### 4. Loss Function Enhancements
Consider adding:
- Energy balance loss (constrain predictions to conserve energy)
- Zone coupling loss (adjacent zones should have realistic temperature relationships)

## Files to Modify/Create

- `tools/data_gen/main.py`: Add ASHRAE case data generation
- `tools/train_surrogate.py`: Add loss function options
- `assets/thermal_surrogate_v2.onnx`: New calibrated model
- `assets/model_metrics_v2.json`: Validation metrics

## References

- Phase 5 details in `docs/ROADMAP.md`
- Phase 5 quickstart in `docs/PHASE5_QUICKSTART.md`
- Training documentation in `docs/PHASE4_TRAINING.md`
